app/scripts/directives/blockEditor.js

```diff
@@ -325,11 +325,19 @@ angular.module('confRegistrationWebApp').directive('blockEditor', function () {
             });
           });
           if (profileCount > 1) {
-            modalMessage.error(
-              'Only one ' +
+            if ($scope.block.profileType == 'GENDER') {
+              var pType = 'Sex';
+            } else {
+              pType =
```

Comment:
`pType` isn't defined before you use it. it's defined in the above If statement, which wouldn't work on the modalMessage. It needed to be defined outside the if statement, so it can be used in the If statement and on modalMessage.

Reply:
After testing, I found that what you have works. But in Typescript (TS), this would throw an error. Typescript is a stricter version of JS, so when possible you want to write like TS when writing JS.

Reply:
Ok I changed it to const like you suggested.